src/components/common/NetworkSelector/index.tsx

```diff
@@ -33,38 +34,45 @@ const NetworkSelector = (): ReactElement => {
   const { configs } = useChains()
   const chainId = useChainId()
   const router = useRouter()
+  const getNetworkLink = useCallback(
+    (shortName: string) => {
+      const shouldKeepPath = [AppRoutes.newSafe.create, AppRoutes.newSafe.load].includes(router.pathname)
 
-  const handleNetworkSwitch = (event: SelectChangeEvent) => {
-    const selectedChainId = event.target.value
-    const newShortName = getShortName(selectedChainId)
+      const route = {
+        pathname: shouldKeepPath ? router.pathname : '/',
+        query: {
+          chain: shortName,
+        } as {
+          chain: string
+          safeViewRedirectURL?: string
+        },
+      }
 
-    if (!newShortName) return
+      if (router.query?.safeViewRedirectURL) {
+        route.query.safeViewRedirectURL = router.query?.safeViewRedirectURL.toString()
+      }
 
-    trackEvent({ ...OVERVIEW_EVENTS.SWITCH_NETWORK, label: selectedChainId })
+      return route
+    },
+    [router],
+  )
 
-    const shouldKeepPath = [AppRoutes.newSafe.create, AppRoutes.newSafe.load].includes(router.pathname)
+  const onChange = (event: SelectChangeEvent) => {
+    event.preventDefault() // Prevent the link click
 
-    const newRoute = {
-      pathname: shouldKeepPath ? router.pathname : '/',
-      query: {
-        chain: newShortName,
-      } as {
-        chain: string
-        safeViewRedirectURL?: string
-      },
-    }
+    const newChainId = event.target.value
+    const shortName = configs.find((item) => item.chainId === newChainId)?.shortName
 
-    if (router.query?.safeViewRedirectURL) {
-      newRoute.query.safeViewRedirectURL = router.query?.safeViewRedirectURL.toString()
+    if (shortName) {
+      trackEvent({ ...OVERVIEW_EVENTS.SWITCH_NETWORK, label: newChainId })
+      router.push(getNetworkLink(shortName))
```

Comment:
Now that each `MenuItem` has a `Link`, you don't need to `push` here as well.

Reply:
Sadly it doesn't work if you use the keyboard. So the link will only work if you open it in a new tab.

Reply:
Intriguing. Just double-checked and it doesn't add two entries to the history 👍